for_curators/sample_local_dataset_upload_dev.py.ipynb

```diff
@@ -0,0 +1,305 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1ff1ec6e",
+   "metadata": {},
+   "source": [
+    "#### This script performs the upload of a local datafile to a given Collection (as identified by its Collection uuid), where the datafile becomes a Dataset accessible via the Data Portal UI. In order to use this script, you must have a Curation API key, you must know the id of the Collection to which you wish to upload the datafile, and you must decide upon a string tag (the `curator_tag`) to use to uniquely identify the resultant Dataset within this Collection going forward. Uploads to a tag that has not been used yet in the given Collection will result in a new Dataset being created. Uploads to a tag for which there already exists a Dataset in the given Collection will result in the existing Dataset being replaced by the new Dataset created from the datafile that you are uploading. You can only add/replace Datasets in private Collections or revision Collections."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "50c6c402",
+   "metadata": {},
+   "source": [
+    "#### <font color='#bc00b0'>Please fill in the required values:</font>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "38afd0cd",
+   "metadata": {},
+   "source": [
+    "<font color='#bc00b0'>(Required) Provide the path to your api key file</font>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 120,
+   "id": "7c8c55fa",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "api_key_file = \"path/to/api-key.txt\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3e84d52e",
+   "metadata": {},
+   "source": [
+    "<font color='#bc00b0'>(Required) Provide the absolute path to the h5ad datafile to upload</font>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 121,
+   "id": "f46e1874",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "filename = \"/absolute/path/to-datafile.h5ad\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "104aea94",
+   "metadata": {},
+   "source": [
+    "<font color='#bc00b0'>(Required) Enter your chosen `curator_tag`, which will serve as a unique identifier _within this Collection_ for the resultant Dataset. **Must possess the '.h5ad' suffix**.</font>\n",
+    "    \n",
+    "_We recommmend using a tagging scheme that 1) makes sense to you, and 2) will help organize and facilitate your \n",
+    "automation of future uploads for adding new Datasets and replacing existing Datasets._"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 122,
+   "id": "b2198be1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "curator_tag = \"arbitrary/tag/choosen-by-you.h5ad\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "463870f1",
+   "metadata": {},
+   "source": [
+    "<font color='#bc00b0'>(Required) Enter the id of the Collection to which you wish to add this datafile as a Dataset</font>\n",
```

Comment:
We might want to mention how to obtain the target collection's UUID, either from the portal UI's URL or curation API endpoint.

Reply:
Added some detail here